src/rebar_app_discover.erl

```diff
@@ -234,9 +236,9 @@ all_app_dirs(LibDirs) ->
 %% @private find the directories for all apps based on their source dirs
 %% Returns the app dir with the respective src_dirs for them, in that order,
 %% for every app found.
--spec all_app_dirs([file:name()], [file:name()]) -> [{file:name(), [file:name()]}].
-all_app_dirs(LibDirs, SrcDirs) ->
-    lists:flatmap(fun(LibDir) -> app_dirs(LibDir, SrcDirs) end, LibDirs).
+-spec all_app_dirs([file:name()], [file:name()], rebar_state:t()) -> [{file:name(), [file:name()]}].
+all_app_dirs(LibDirs, SrcDirs, State) ->
+    lists:flatmap(fun(LibDir) -> app_dirs(LibDir, SrcDirs, State) end, LibDirs).
```

Comment:
Th eproblem is that these calls might then not be correct anymore if the project uses this functionality. Furthermore, some of the function interfaces are conflicting, e.g. find_apps/2. 

We could introduce a new API and keep the current one for backwards compatibility. But that would generate bad APIs. I thought about this myself quite some bit, but didn't really come up with a good solution.

Reply:
I'll try to see if I can find usages on github to get a better understanding on how widely used this API is.

Reply:
I used the GIthub code search and only found 
https://github.com/alinpopa/rebar3-localdep-plugin
using `rebar_app_discover:find_apps/2` that would break. The repo has not been updated in the last 4 years. I think it's safe to not provide backwards compatibility for those APIs.

(see https://github.com/search?l=Erlang&q=rebar_app_discover%3A&type=Code)